torch/utils/_pytree.py

```diff
@@ -964,11 +964,16 @@ def unflatten(self, leaves: Iterable[Any]) -> PyTree:
         return unflatten_fn(child_pytrees, self.context)
 
 
+@dataclasses.dataclass(frozen=True, repr=False)
```

Comment:
Does this affect the inherited `__eq__` in any way?

Reply:
`@dataclasses.dataclass` has `eq=True` by default, and will create a semantically equivalent `LeafSpec.__eq__` as `TreeSpec.__eq__`. Lemme use `eq=False` explicitly here to demand reuse. In theory one can also provide a faster `LeafSpec.__eq__` that's type based only (rather than comparing individual fields, in the generated `__eq__`).

Reply:
thanks for checking